activation/activation.go

```diff
@@ -451,6 +455,20 @@ func (b *Builder) run(ctx context.Context, sig *signing.EdSigner) {
 				return
 			case <-time.After(b.poetRetryInterval):
 			}
+		case errors.Is(err, ErrInvalidInitialPost):
+			// delete the existing db post
+			// call build initial post again
+			b.logger.Error("initial post is no longer valid. regenerating initial post")
+			if err := b.nipostBuilder.ResetState(sig.NodeID()); err != nil {
+				b.logger.Error("failed to reset nipost builder state", zap.Error(err))
+			}
+			if err := nipost.RemoveChallenge(b.localDB, sig.NodeID()); err != nil {
+				b.logger.Error("failed to discard challenge", zap.Error(err))
+			}
+			if err := nipost.RemovePost(b.localDB, sig.NodeID()); err != nil {
+				b.logger.Error("failed to remove existing post from db", zap.Error(err))
+			}
+			goto BUILD
```

Comment:
I think you could make it work without a `goto`.  How about putting the:

```go
for {
	err := b.buildInitialPost(ctx, sig.NodeID())
	if err == nil {
		break
	}
	b.logger.Error("failed to generate initial proof:", zap.Error(err))
	currentLayer := b.layerClock.CurrentLayer()
	select {
	case <-ctx.Done():
		return
	case <-b.layerClock.AwaitLayer(currentLayer.Add(1)):
	}
}
```

into a function and calling it here? The certification part (`poet.Certify(...)`) doesn't need to be repeated.